### PR TITLE
Fix trial_end type to allow special value 'now' 

### PR DESCRIPTION
**Pull request by acoll**
https://stripe.com/docs/api/node

The api allows the special value of 'now' to be passed for trial_end.

### CODE REVIEW

**Comment by CodeAnimal** [comment]
Looks good to me, although [L1333](https://github.com/acoll/DefinitelyTyped/blob/459189c25ab2da839da5158532bbea2d42ef90df/types/stripe/index.d.ts#L1333) is redundant as it isn't a valid [creation option](https://stripe.com/docs/api/node#create_customer) for a customer any more. 👍 

I'll update the customer creation options at some point though, so for now this looks good.

**Review by CodeAnimal — APPROVED** [review]

**Review by CodeAnimal — CHANGES_REQUESTED** [review]
Actually, can we change the `string` type to `"now"` so it explicitly only allows that word, not just any string.

**Comment by CodeAnimal** [comment]
And can you then add some tests too, please.

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -1330,7 +1330,7 @@ declare namespace Stripe {
              * override the default trial period of the plan the customer is being subscribed to. The special value now can be provided to
              * end the customer's trial immediately. Only applies when the plan parameter is also provided.
              */
-            trial_end?: number;
+            trial_end?: number | string;
```
Comment (LinusU) [inline types/stripe/index.d.ts]: As noted, please change this to `number | "now";` 👍

**Review by LinusU — APPROVED** [review]
Perfect 👍

**Review by CodeAnimal — APPROVED** [review]
Good work 👍

**Pull request merged**